stix_shifter/stix_translation/src/modules/guardium/query_constructor.py

```diff
@@ -0,0 +1,524 @@
+                return "{expr1} {operator} {expr2}".format(expr1=self._parse_expression(expression.observation_expression.expr1),
+                                                           operator=operator,
+                                                           expr2=self._parse_expression(expression.observation_expression.expr2, expression.qualifier))
+            else:
+                return self._parse_expression(expression.observation_expression.comparison_expression, expression.qualifier)
+        elif isinstance(expression, CombinedObservationExpression):
+            operator = self.comparator_lookup[expression.operator]
+            expression_01 = self._parse_expression(expression.expr1)
+            expression_02 = self._parse_expression(expression.expr2)
+            if expression_01 and expression_02:
+                return "({}) {} ({})".format(expression_01, operator, expression_02)
+            elif expression_01:
+                return "{}".format(expression_01)
+            elif expression_02:
+                return "{}".format(expression_02)
+            else:
+                return ''
+        elif isinstance(expression, Pattern):
+            return "{expr}".format(expr=self._parse_expression(expression.expression))
+        else:
+            raise RuntimeError("Unknown Recursion Case for expression={}, type(expression)={}".format(
+                expression, type(expression)))
+
+    def parse_expression(self, pattern: Pattern):
+        logging.debug("Parse Expression: patten --->")
+        logging.debug(pattern)
+        return self._parse_expression(pattern)
+
+
+def translate_pattern(pattern: Pattern, data_model_mapping, options):
+    logging.debug("\nTranslate Pattern: " + str(pattern))
+    logging.debug("Entered translate_pattern------\nTranslate Pattern: " + str(pattern))
+    logging.debug("Data model mapping: " + str(data_model_mapping))
+    # Converting query object to datasource query
+    # timerange set to 24 hours for Guardium; timerange is provided in minutes (as delta)
+    timerange = 24 * 60 * 60
+    parsed_stix = parse_stix(pattern,timerange)
+
+    logging.debug("Parsed_stix: " + str(parsed_stix))
+    logging.debug("Options supplied to query Constructer - translate pattern --")
+    logging.debug(options)
+#
+    guardiumQueryTranslator = QueryStringPatternTranslator(pattern, data_model_mapping)
+    reportCall = guardiumQueryTranslator.translated
+
+    # Add space around START STOP qualifiers
+    reportCall = re.sub("START", "START ", reportCall)
+    reportCall = re.sub("STOP", " STOP ", reportCall)
+    logging.debug("reportCall -->")
+    logging.debug(reportCall)
+#Subroto: I did not change the code much just adapted to get the report parameters
+#Subroto: added code to support report search parameters are "and" when sent to Guardium
+#
+#   Limit recursive call to build an array of report parameters
+    sys.setrecursionlimit(500)
```

Comment:
We already bump up the recursion limit in `stix_translation.py`. Is there a reason you are decreasing it here?